File: 2668-form-smallest-number-from-two-digit-arrays/form-smallest-number-from-two-digit-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var minNumber = function (nums1, nums2) {
    let min1 = 11;
    let freq = new Array(10);
    let freq2 = new Array(10);
    for (let i of nums1) {
        min1 = Math.min(min1, i);
        freq[i] = true;
    }
    let min2 = 11;
    for (let i of nums2) {
        min2 = Math.min(i, min2);
        freq2[i] = true;
    }
    for (let i = 1; i < 10; i++) {
        if (freq[i] && freq2[i]) return i;
    }
    if (min1 < min2) return min1 * 10 + min2;

    return min2 * 10 + min1;
};

File: 2668-form-smallest-number-from-two-digit-arrays/form-smallest-number-from-two-digit-arrays.java
class Solution {
    public int minNumber(int[] nums1, int[] nums2) {
        int min1 = 11;
        boolean freq [] = new boolean[10];
        boolean freq2 [] = new boolean[10];
        for(int i : nums1){
            min1 = Math.min(min1,i);
            freq[i] = true;
        }
        int min2 = 11;
        for(int i : nums2){
            min2 = Math.min(i,min2);
            freq2[i] = true;
        }
        for(int i = 1; i < 10; i++){
            if(freq[i] && freq2[i]) return i;
        }
        if(min1 < min2) return min1*10+min2;
        
        return min2*10+min1;
    }
}